actioncable/app/javascript/action_cable/adapters.js

```diff
@@ -1,4 +1,16 @@
+const logger = typeof globalThis !== 'undefined'
+                ? globalThis.console
+                : typeof self !== 'undefined' // For IE Support
+                ? self.console
+                : undefined
+
+const WebSocket = typeof globalThis !== 'undefined'
```

Comment:
Wouldn't checking for `typeof WebSocket` be enough?

```js
const WebSocket = typeof WebSocket !== 'undefined' ? WebSocket : undefined
```

Similarly for `console`.

Reply:
I don't think that would work, as a naming error. 
```ts
const WebSocket = typeof WebSocket !== 'undefined' ? WebSocket : undefined
```
should be functionally the same as
```
const WebSocket = WebSocket
```

Reply:
> functionally the same as

Depends on the runtime; in Node, `typeof Unknown === 'undefined'` while `Unknown #=> ReferenceError`.

We use `typeof ..` check in [anycable-client](https://github.com/anycable/anycable-client/blob/master/packages/core/websocket/index.js#L11); works fine in browsers (incl. shared workers), Node, React Native,